### PR TITLE
Add builtin documentation for graphics

### PR DESCRIPTION
**Pull request by juliaogris**
Add builtin documentation for graphics, part 1. We still need to
document `dash`, `linecap`, `text` and `font`, but its a big section
so we'll leave it for later.

Here are graphics in the context of builtin docs:

✅ I/O
✅ Types
✅ Map
✅ Program Control
✅ Conversion
✅ Error
✅ String
✅ Random
✅ Math
👉 Graphics
⬜️ Event Handlers

---

Preview link: https://github.com/foxygoat/evy/blob/builtin-ui-1/docs/builtins.md#graphics

### CODE REVIEW

File: docs/builtins.md [hunk 1]
```diff
@@ -1256,3 +1256,463 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+the Cartesian coordinate system used in mathematics. The horizontal
+dimension is called the **x-axis**, and the vertical dimension is
+called the **y-axis**.
+
+A point on the canvas is defined by its **x** and **y coordinates**,
+which are written as `x y`. For example, the point `30 60` has an
+x-coordinate of 30 and a y-coordinate of 60. It is located 30 units
+from the left edge of the canvas and 60 units from the bottom edge.
+
+<img width="300" alt="Catesian coordinates with x and y axes and point (30, 60)" src="img/coords.png">
```
Comment (camh-) [inline docs/builtins.md:1275]: s/Catesian/Cartesian/

Unless you really meant :cat:

File: docs/builtins.md [hunk 2]
```diff
@@ -1256,3 +1256,463 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+
+[Evy source](https://evy.dev/#content=H4sIAAAAAAAAE0WRwW6EIBCG7/MUE86NAV1w8dqkL7HxYJFtSVE2rmnXNn33DgL2NDMfA///h7fFjWCCDwuyVz+YDwZfblzfkVcnAO9mixwF5zCFz9jyxIhQC7fgN7zsQ48XfUZRydTEOfG8tL8SYSVRn/sIcokc4BrmFX/u7tt2J7wOk/Nbx17cMuBzGO0TTmEO99tgLPtNXrREAat9rMgeLCFBMKPtQGWJM0ioOVJQp3gJqY48hKUma7VG2VJtRK47z0t0qGKeus3LLSrRF56laomqzvq72OFBYK2L1e6fC5JQJVXXRJ4/Z7EjO/yTV+MW4+kC/AGSFdVbwgEAAA==).
+
+The canvas ranges from coordinates `0 0` to `100 100`. The center of the
```
Comment (camh-) [inline docs/builtins.md:1279]: s/center/centre/

File: docs/builtins.md [hunk 3]
```diff
@@ -1256,3 +1256,463 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+canvas has the coordinates `50 50`.
+
+### `move`
+
+`move` sets the position of the drawing pen to the given coordinates.
```
Comment (camh-) [inline docs/builtins.md:1284]: The "drawing pen" (aka cursor) should be described in the Graphics preamble above. Also sometimes below you say "cursor position" and others "position of the drawing pen" and others "position of the pen". You should probably pick one and stick with it. "cursor" is more technical and probably doesn't make sense to new comers I guess.

File: docs/builtins.md [hunk 4]
```diff
@@ -1256,3 +1256,463 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+
+Output
+
+<img width="300" alt="black circle in the canvas center" src="img/circle.png">
+
+#### Reference
+
+    circle radius:num
+
+The `circle` function draws a circle with the given `radius` centered
```
Comment (camh-) [inline docs/builtins.md:1379]: s/centered/centred/

File: docs/builtins.md [hunk 5]
```diff
@@ -1256,3 +1256,463 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+
+The `width` function sets the thickness of the _stroke_ to the given `n`
+units. The stroke is the visible line that is drawn when using the
+`line` function or any other shape function after setting
+`fill "none"`.
+
+### `color`
+
+`color` changes the color of the drawing pen. All
+[CSS (Cascading StyleSheets) color values](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value) are
```
Comment (camh-) [inline docs/builtins.md:1415]: s/StyleSheets/Style Sheets/

File: docs/builtins.md [hunk 6]
```diff
@@ -1256,3 +1256,463 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+supported. You can start by using the simpler
+[named CSS colors](https://developer.mozilla.org/en-US/docs/Web/CSS/named-color),
+such as `"red"`, `"darkmagenta"`, and `"lightseagreen"`.
```
Comment (camh-) [inline docs/builtins.md:1418]: :pineapple: I like springgreen - it is my terminal text colour (has been for a long long time).

File: docs/builtins.md [hunk 7]
```diff
@@ -1256,3 +1256,463 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+    color c:string
+
+The `color` function changes the color of the _stroke_ and the _fill_ to
+the given CSS color string `c`. Evy supports all
+[CSS color values](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value),
+including semi-transparent ones. For example, the following code changes
+the color to a shade of red that is 60% opaque:
+`color "hsl(0deg 100% 50% / 60%)"`.
+
+_named CSS colors_, such as `"red"`, `"darkmagenta"`, and
```
Comment (camh-) [inline docs/builtins.md:1442]: "named" should be capitalised

File: docs/builtins.md [hunk 8]
```diff
@@ -1256,3 +1256,463 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+`"lightseagreen"`, are a simpler way of specifying
+common colors. For a complete list of named CSS colors, see the
+[Mozilla Developer documentation](https://developer.mozilla.org/en-US/docs/Web/CSS/named-color).
+
+### `colour`
+
+See [`color`](#color).
```
Comment (camh-) [inline docs/builtins.md:1449]: Text: `colour` is an alternate spelling of `color`. See [`color`](#color).

File: docs/builtins.md [hunk 9]
```diff
@@ -1256,3 +1256,463 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+#### Reference
+
+    width n:num
+
+The `width` function sets the thickness of the _stroke_ to the given `n`
+units. The stroke is the visible line that is drawn when using the
+`line` function or any other shape function after setting
+`fill "none"`.
+
+### `color`
```
Comment (camh-) [inline docs/builtins.md:1412]: I'm torn - `color` is the US spelling, so I feel it should be the "alternate" one, and the word "colour" should be used in the text. But I assume you have deliberately decided to use american spelling, so perhaps also forget about my comments about center/centre

**Review by camh- — APPROVED** [review]
:green_square: Some simple fixup comments and suggestions. All very good.

File: docs/builtins.md (hunk 1 above)
Comment (juliaogris) [reply docs/builtins.md:1275]: thanks!

File: docs/builtins.md [hunk 10]
```diff
@@ -1256,3 +1256,463 @@ x-axis and the ray from the origin to the point `x y`. More formally,
 it returns the arc tangent of `y/x` for given arguments `y` and `x`,
 using the signs of the two to determine the quadrant of the return
 value.
+
+## Graphics
+
+`evy` on the web outputs drawing commands to a drawing area
+called **canvas**.
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply docs/builtins.md:1263]: ok.

File: docs/builtins.md (hunk 5 above)
Comment (juliaogris) [reply docs/builtins.md:1415]: thanks.

File: docs/builtins.md (hunk 6 above)
Comment (juliaogris) [reply docs/builtins.md:1418]: let's have a little spring green....

File: docs/builtins.md (hunk 7 above)
Comment (juliaogris) [reply docs/builtins.md:1442]: done, thanks.

File: docs/builtins.md (hunk 8 above)
Comment (juliaogris) [reply docs/builtins.md:1449]: thanks, done.

File: docs/builtins.md (hunk 3 above)
Comment (juliaogris) [reply docs/builtins.md:1284]: ok, done.
I kept cursor for reference and pen for descriptions and added an explanation to the intro.

**Review by camh- — COMMENTED** [review]
I've made a couple of separate comments about specifying initial values for pen, color, width, fill (and I think I forgot stroke). Maybe that should go up front? Or it could go in each builtin's reference section that changes that value.

Also, is the ordering of the builtins deliberate or just incidental? I found it a little odd that `fill` was right at the end (I was looking for it and thought I missed it). Perhaps the "context" functions could go together and the drawing functions together?

File: docs/builtins.md [hunk 11]
```diff
@@ -1256,3 +1256,475 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+
+#### Reference
+
+    poly xy:[]num...
+
+The `poly` function draws polylines and polygons for the given
+coordinates. A polyline is a sequence of connected line segments, and a
+polygon is a closed polyline.
+
+The `poly` function takes a variadiac number of arguments of type
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply docs/builtins.md:1625]: done, thanks.

File: docs/builtins.md [hunk 12]
```diff
@@ -1256,3 +1256,475 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+`[]num`. Each argument has to be a number array with two elements
+`[x y]`. The first element representing the x coordinate and the second
+the y coordinate of a vertex in the polyline or polygon. If the array
+does not have two elements, a fatal runtime error occurs. This means
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply docs/builtins.md:1629]: done, agreed, better.

File: docs/builtins.md [hunk 13]
```diff
@@ -1256,3 +1256,475 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+that the `poly` function can be called as follows:
+
+    poly [x1 y1] [x2 y2] [x3 y3] // ...
+
+Use `fill "none"` to draw a line without filling. To draw a filled
+polygon, make sure that the first and last coordinates are the same.
+
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply docs/builtins.md:1636]: done, thanks.

File: docs/builtins.md [hunk 14]
```diff
@@ -1256,3 +1256,475 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+Output
+
+<img width="300" alt="two black rectangles with touching vertices" src="img/rects.png">
+
+#### Reference
+
+    rect width:num height:num
+
+The `rect` function draws a rectangle with the given `width` and
+`height` at the current cursor position. The cursor position is then
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply docs/builtins.md:1365]: done.

File: docs/builtins.md [hunk 15]
```diff
@@ -1256,3 +1256,475 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+<img width="300" alt="diagonal lines of different width" src="img/width.png">
+
+#### Reference
+
+    width n:num
+
+The `width` function sets the thickness of the _stroke_ to the given `n`
+units. The stroke is the visible line that is drawn when using the
+`line` function or any other shape function after setting
+`fill "none"`.
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply docs/builtins.md:1461]: done, thanks.

File: docs/builtins.md [hunk 16]
```diff
@@ -1256,3 +1256,475 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+`[]num`. Each argument has to be a number array with two elements
+`[x y]`. The first element representing the x coordinate and the second
+the y coordinate of a vertex in the polyline or polygon. If the array
+does not have two elements, a fatal runtime error occurs. This means
+that the `poly` function can be called as follows:
+
+    poly [x1 y1] [x2 y2] [x3 y3] // ...
+
+Use `fill "none"` to draw a line without filling. To draw a filled
+polygon, make sure that the first and last coordinates are the same.
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply docs/builtins.md:1635]: done.

File: docs/builtins.md [hunk 17]
```diff
@@ -1256,3 +1256,475 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+Output
+
+<img width="300" alt="a red sqaures and a blue square with red outline" src="img/fill.png">
+
+#### Reference
+
+    fill c:string
+
+The `fill` function sets the color of the _fill_ to the given string
+argument `c`. The fill is the interior of a shape.
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply docs/builtins.md:1730]: done.

File: docs/builtins.md [hunk 18]
```diff
@@ -1256,3 +1256,475 @@ x-axis and the ray from the origin to the point `x y`. More formally,
...
+from the left edge of the canvas and 60 units from the bottom edge.
+
+<img width="300" alt="Cartesian coordinates with x and y axes and point (30, 60)" src="img/coords.png">
+
+[Evy source](https://evy.dev/#content=H4sIAAAAAAAAE0WRwW6EIBCG7/MUE86NAV1w8dqkL7HxYJFtSVE2rmnXNn33DgL2NDMfA///h7fFjWCCDwuyVz+YDwZfblzfkVcnAO9mixwF5zCFz9jyxIhQC7fgN7zsQ48XfUZRydTEOfG8tL8SYSVRn/sIcokc4BrmFX/u7tt2J7wOk/Nbx17cMuBzGO0TTmEO99tgLPtNXrREAat9rMgeLCFBMKPtQGWJM0ioOVJQp3gJqY48hKUma7VG2VJtRK47z0t0qGKeus3LLSrRF56laomqzvq72OFBYK2L1e6fC5JQJVXXRJ4/Z7EjO/yTV+MW4+kC/AGSFdVbwgEAAA==).
+
+The canvas ranges from coordinates `0 0` to `100 100`. The center of the
+canvas has the coordinates `50 50`.
+
+Shapes are drawn on the canvas using a _pen_. The pen has a position and
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply docs/builtins.md:1282]: done, thanks.

**Review by camh- — APPROVED** [review]
:green_square: LGTM (still)

**Pull request merged**